src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderSettingsDaoTest.java

```diff
@@ -0,0 +1,248 @@
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Category(Unit.class)
+@EmbeddedDatabaseTest
+public class SnapshotBuilderSettingsDaoTest {
+  private static final Logger logger = LoggerFactory.getLogger(DatasetBucketDaoTest.class);
+
+  @Autowired private DaoOperations daoOperations;
+  @Autowired private ProfileDao profileDao;
+  @Autowired private GoogleResourceDao resourceDao;
+  @Autowired private DatasetDao datasetDao;
+  @Autowired private SnapshotBuilderSettingsDao snapshotBuilderSettingsDao;
+
+  private BillingProfileModel billingProfile;
+  private GoogleProjectResource projectResource;
+  private Dataset dataset;
+  private SnapshotBuilderSettings snapshotBuilderSettings;
+
+  @Before
+  public void setup() throws IOException {
+    BillingProfileRequestModel profileRequest = ProfileFixtures.randomBillingProfileRequest();
+    billingProfile = profileDao.createBillingProfile(profileRequest, "testUser");
+
+    projectResource = ResourceFixtures.randomProjectResource(billingProfile);
+    UUID projectId = resourceDao.createProject(projectResource);
+    projectResource.id(projectId);
+
+    dataset = daoOperations.createMinimalDataset(billingProfile.getId(), projectId);
+    snapshotBuilderSettings =
+        snapshotBuilderSettingsDao.upsertSnapshotBuilderSettingsByDataset(
+            dataset.getId(), SAMPLE_SNAPSHOT_BUILDER_SETTINGS);
+  }
+
+  @After
+  public void teardown() {
+    try {
+      snapshotBuilderSettingsDao.delete(dataset.getId());
+    } catch (Exception ex) {
+      logger.error(
+          "[CLEANUP] Unable to delete snapshot builder settings for dataset {}", dataset.getId());
+    }
+    try {
+      datasetDao.delete(dataset.getId());
+    } catch (Exception ex) {
+      logger.error("[CLEANUP] Unable to delete dataset {}", dataset.getId());
+    }
+    try {
+      resourceDao.deleteProject(projectResource.getId());
+    } catch (Exception ex) {
+      logger.error(
+          "[CLEANUP] Unable to delete entry in database for projects {}", projectResource.getId());
+    }
+    try {
+      profileDao.deleteBillingProfileById(billingProfile.getId());
+    } catch (Exception ex) {
+      logger.error("[CLEANUP] Unable to billing profile {}", billingProfile.getId());
+    }
+  }
+
+  @Test
+  public void getSnapshotBuilderSettingsReturnsSettings() {
+    assertThat(
+        "Snapshot builder settings should be the same as the example",
+        snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(dataset.getId()),
+        equalTo(SAMPLE_SNAPSHOT_BUILDER_SETTINGS));
+  }
+
+  @Test
+  public void getSnapshotBuilderSettingsForDatasetThatDoesNotExistErrors() {
+    assertThrows(
+        NotFoundException.class,
+        () -> snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(UUID.randomUUID()));
```

Comment:
Sonar is going to flag this since `UUID.randomUUID()` can throw.
```suggestion
   var id = UUID.randomUUID();
    assertThrows(
        NotFoundException.class,
        () -> snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(id));
```